A little cleanup of the README and prettifying console output

GreenGremlin opened the pull request:
Mostly these changes are just cosmetic, with a bugfix here or there.

### Screenshots

Start-up

![image](https://user-images.githubusercontent.com/647452/41645108-d8a4a776-7424-11e8-8cdf-106ab8321c3e.png)

Generating response

![image](https://user-images.githubusercontent.com/647452/41645174-071adbfc-7425-11e8-92aa-946e91f8b9b7.png)

Sending SAML response

![image](https://user-images.githubusercontent.com/647452/41645217-24863c5e-7425-11e8-9df0-7ac9b25f9e0f.png)

GreenGremlin commented on app.js:
```diff
@@ -34,37 +35,45 @@ const IDP_PATHS = {
-  return cryptTypes[type] && cryptTypes[type].test(value);
-}
-
-function bufferFromString(value) {
```
This was unused

mcguinness replied:
👍

GreenGremlin commented on app.js:
```diff
@@ -73,13 +82,9 @@ function resolveFilePath(filePath) {
-  ['.', __dirname].forEach(function (base) {
-    possiblePath = path.resolve(base, filePath);
-    if (fs.existsSync(possiblePath)) {
-      return possiblePath;
```
Returning in a `forEach` doesn't actually do anything, so this didn't actually work.

GreenGremlin commented on app.js:
```diff
@@ -110,27 +114,67 @@ function getHashCode(str) {
 function processArgs(args, options) {
   var baseArgv;
-  console.log();
-  console.log('loading configuration...');
```
This seemed unnecessary, so I removed it. I don't mind adding it back, if you disagree.

mcguinness replied:
i'm fine with it

GreenGremlin commented on app.js:
```diff
@@ -110,27 +114,67 @@ function getHashCode(str) {
   if (options) {
     baseArgv = yargs(args).config(options);
   } else {
-    baseArgv = yargs(args).config('settings', function(settingsPathArg) {
```
This PR is branched off my other PR

GreenGremlin commented on app.js:
```diff
@@ -110,27 +114,67 @@ function getHashCode(str) {
     .usage('\nSimple IdP for SAML 2.0 WebSSO & SLO Profile\n\n' +
         'Launches an IdP web server that mints SAML assertions or logout responses for a Service Provider (SP)\n\n' +
         'Usage:\n\t$0 -acs {url} -aud {uri}')
+    .alias({h: 'help'})
```
Why not :)

GreenGremlin commented on app.js:
```diff
@@ -268,7 +312,7 @@ function processArgs(args, options) {
       }
     })
-    .example('\t$0 --acs http://acme.okta.com/auth/saml20/exampleidp --aud https://www.okta.com/saml2/service-provider/spf5aFRRXFGIMAYXQPNV', '')
+    .example('$0 --acsUrl http://acme.okta.com/auth/saml20/exampleidp --audience https://www.okta.com/saml2/service-provider/spf5aFRRXFGIMAYXQPNV', '')
```
I figured it was better to use full option names. Honestly, I'd rather see this named something more verbose like `--assertionConsumerServiceUrl`.

mcguinness replied:
lets stick with the short names (ok with +Url alias), i see the value in being verbose but our team uses this from CLI a lot and the short names are easier to deal with.

GreenGremlin replied:
I'm only using the full option name for the example, the short names still work. I think using the full names in the example makes the example easier to understand. If you disagree, I can revert the example back to the short option.

GreenGremlin commented on app.js:
```diff
@@ -299,37 +343,47 @@ function processArgs(args, options) {
-  console.log();
-
-  console.log();
+  console.log(dedent(chalk`
```
Pretty colors and more readable source

mcguinness replied:
This is an awesome enhancement, its been on my list of things to add for a long time just haven't had time.  Thanks for the feature!

GreenGremlin commented on app.js:
```diff
@@ -110,27 +114,67 @@ function getHashCode(str) {
+    return chalk`${
+      value.toString()
+        .replace(/-----.+?-----|\n/g, '')
+        .substring(0, 80)
```
Before this change key / cert was printed as something like `<Buffer 2d 2d 2d 2d 2d 42 ... >`

mcguinness replied:
nice addition

GreenGremlin commented on app.js:
```diff
@@ -110,27 +114,67 @@ function getHashCode(str) {
+
+function prettyPrintXml(xml, indent) {
+  // This works well, because we format the xml before applying the replacements
+  const prettyXml = xmlFormat(xml, {indentation: '  '})
```
I spent a little time searching for a library to do this, but in the end it was much more simple to just use a couple regexes.

mcguinness replied:
cool, we can go with this for now until need changes for a lib.

GreenGremlin commented on app.js:
```diff
@@ -299,37 +343,47 @@ function processArgs(args, options) {
+    Encrypt Assertion:
+      {blue ${argv.encryptAssertion}}
+    Authentication Context Class Reference:
+      {blue ${argv.authnContextClassRef || UNDEFINED_VALUE}}
```
I'm using `UNDEFINED_VALUE` as a fallback because I think

```
Authentication Context Declaration:
  None
```

looks better than 

```
Authentication Context Declaration:
  undefined
```

I'm open to using a different fallback, if you don't like "none".

mcguinness replied:
👍

mcguinness commented on app.js:
```diff
@@ -34,37 +35,45 @@ const IDP_PATHS = {
 function resolveFilePath(filePath) {
+  if (filePath.startsWith('saml-idp/')) {
+    // Allows file path options to files included in this package, like config.js
+    const resolvedPath = require.resolve(filePath.replace(/^saml\-idp\//, `${__dirname}/`));
```
the local repo name (e.g. `git clone`) can be something different than `saml-idp` which would be more constant for `npm install`.   I'm sure there is an environment variable for where the root is located so we are not hard-coding `saml-idp`.

GreenGremlin replied:
I'm actually using `__dirname` to resolve the file path here, which I think is what you're asking for. The primary purpose for this change is so that the default value for `--configFile` doesn't print as a full, resolved path. All paths should resolve just as they did before, regardless of wether the package is installed from npm or cloned using git.

You'll notice README.md contains an absolute path (`/Users/karl/src/saml-idp/config.js`) which appears to have come from the output of `node ./app --help` from Karl's src directory. With this change, `node ./app --help` will show the default value for `--configFile` as `saml-idp/config.js`.

As an added benefit, users can pass file paths to files in this package using the `saml-idp` root name. For example, `saml-idp --key saml-idp/idp-private-key.pem --cert  saml-idp/idp-public-cert.pem` will start the IDP using the packaged key / cert pair, something that wasn't previously possible if you had installed via `npm install saml-idp`. Or at least you would have had to specify the full path, something like `--key ./node_modules/saml-idp/idp-private-key.pem`.

mcguinness commented on app.js:
```diff
@@ -239,7 +283,7 @@ function processArgs(args, options) {
         description: 'Path to a SAML attribute config file',
         required: true,
-        default: require.resolve('./config.js'),
+        default: 'saml-idp/config.js',
```
same comment as above, we shouldn't hard-code `saml-idp` path and use a process variable instead

mcguinness commented:
Can you rebase this on master without the README changes (I already added these).  I would like to merge this

GreenGremlin commented:
> Can you rebase this on master without the README changes (I already added these). I would like to merge this

I rebased this on master. There are still some README changes, but I think they're still relevant. Let me know if you see any issues and I'll address them.
